src/Driver/PgSQL/Connection.php

```diff
@@ -67,7 +69,18 @@ public function prepare(string $sql): Statement
 
     public function query(string $sql): Result
     {
-        return $this->prepare($sql)->execute();
+        if (@pg_send_query($this->connection, $sql) !== true) {
+            throw new Exception(pg_last_error($this->connection));
+        }
+
+        $result = @pg_get_result($this->connection);
+        assert($result !== false);
```

Comment:
What should happen when a query returns 0 records? Do we still get a `PgSql\Result` instance?

Reply:
Yes. An empty result is still a result.

Also, that case is really well covered by functional tests. 😀